### ISSUE TITLE
batch parallel decoding

### ISSUE THREAD

**Issue by handsomelys** [gpt-omni/mini-omni#108]
May I ask where the "batch parallel decoding" mentioned in the paper is specifically reflected in the code base? I don't quite understand how this technology is implemented.

**Comment by vra** [comment]
@handsomelys Hi, Please check the related code snippets below:
1. Obtain audio logits from sample0, text logits from sample1: <https://github.com/gpt-omni/mini-omni/blob/main/litgpt/generate/base.py#L155-L157>
2. construct batch size=2's inputs for next iteration: <https://github.com/gpt-omni/mini-omni/blob/main/litgpt/generate/base.py#L379-L388>

**Comment by handsomelys** [comment]
> 2\. https://github.com/gpt-omni/mini-omni/blob/main/litgpt/generate/base.py#L379-L388
Thanks!
